pymeasure/display/console.py

```diff
@@ -164,118 +181,105 @@ def __init__(self,
             # Ignore and assume it is a valid level string
-            log_level = args['log_level']
+            log_level = self.args['log_level']
         self.log_level = log_level
         log.setLevel(self.log_level)
         self.log.setLevel(self.log_level)
 
-        if args['sequence_file'] is not None:
+        if self.args['sequence_file'] is not None:
             raise NotImplementedError("Sequencer not yet implemented")
 
         # Set procedure parameters
         parameter_values = {}
 
-        if args['use_result_file'] is not None:
+        if self.args['use_result_file'] is not None:
             # Special case set parameters from log file
-            results = Results.load(args['use_result_file'])
+            results = Results.load(self.args['use_result_file'])
             for name in results.parameters:
                 parameter_values[name] = results.parameters[name].value
         else:
-            for name in args:
+            for name in self.args:
                 opt_name = name.replace("_", "-")
                 if not (opt_name in self.parser.special_options):
-                    parameter_values[name] = args[name]
-
-        procedure.set_parameters(parameter_values)
-        if (progressbar and not args['no_progressbar']):
-            progressbar.streams.wrap_stderr()
-            self.bar = progressbar.ProgressBar(max_value=100,
-                                               prefix='{variables.status}: ',
-                                               variables={'status': "Unknown"})
+                    parameter_values[name] = self.args[name]
+
+        # Handle Ctrl+C nicely
+        signal.signal(signal.SIGINT, lambda sig, _: self.abort())
+
+    def get_filename(self, directory):
+        """ Return filename for logging.
+
+        User can override this method to define their own filename
+        """
+        if self.filename is not None:
+            return os.path.join(directory, self.filename)
         else:
-            self.bar = None
-        scribe = console_log(self.log, level=self.log_level)
-        scribe.start()
+            return unique_filename(directory)
+
+    def queue(self, procedure=None):
+        raise NotImplementedError("Abstract method, must be overridden by the child class.")
 
-        results = Results(procedure, self.get_filename(self.directory))
-        log.debug("Set up Results")
+    def queued(self):
+        raise NotImplementedError("Multiple queued experiments not yet implemented")
 
-        self._worker = Worker(results, log_queue=scribe.queue, log_level=self.log_level)
-        log.info("Created worker for procedure {}".format(self.procedure_class.__name__))
+    def abort_returned(self):
+        self._terminate("Running experiment has returned after an abort")
+
+    def finished(self):
+        self._terminate("Running experiment has finished", 100)
+
+    def failed(self):
+        self._terminate("Running experiment has failed")
+
```

Comment:
Why these method names no longer have '_' prefix conventionally indicating that this is an internal method ?

Reply:
This change is also to bring it inline with the pattern of [`ManagedWindow`](https://github.com/mcdo0486/pymeasure/blob/bafb2b3ae37b2feeccbf560acc468c05bbfb8d6d/pymeasure/display/windows/managed_window.py#L490) which has those functions in the 'public' format, or no leading underscore. That's pretty much it. In console mode at the moment only "abort" button can be thought to be called by "ctrl-c".